File: python-ds-practice/34_same_frequency/same_frequency.py
def same_frequency(num1, num2):
    # Convert numbers to strings
    str_num1 = str(num1)
    str_num2 = str(num2)
    
    # Create dictionaries to count the frequency of each digit
    count_num1 = {}
    count_num2 = {}

    # Count the digits for num1
    for digit in str_num1:
        count_num1[digit] = count_num1.get(digit, 0) + 1

    # Count the digits for num2
    for digit in str_num2:
        count_num2[digit] = count_num2.get(digit, 0) + 1

    # Compare the dictionaries
    return count_num1 == count_num2

# Test cases
print(same_frequency(551122, 221515))  # Expected: True
print(same_frequency(321142, 3212215)) # Expected: False
print(same_frequency(1212, 2211))      # Expected: True

File: prob-freq-count-pointers/same-frequency.js
function sameFrequency(num1, num2) {
    // Helper function to get the count of each digit in a number
    const getDigitCount = (num) => {
        const count = {};
        // Convert number to string and iterate over each character
        for (let digit of String(num)) {
            // Increment the count of the digit in the object
            count[digit] = (count[digit] || 0) + 1;
        }
        return count;
    }

    // Get the digit counts for both numbers
    const count1 = getDigitCount(num1);
    const count2 = getDigitCount(num2);

    // Compare the length of the keys (unique digits) in both counts
    if (Object.keys(count1).length !== Object.keys(count2).length) {
        return false;
    }

    // Compare the count of each digit in both counts
    for (let key in count1) {
        if (count1[key] !== count2[key]) {
            return false;
        }
    }

    return true;
}

module.export = {
    sameFrequency
};

